### ISSUE TITLE
Importing disables Notebook stdout

### ISSUE THREAD

**Issue by mwaskom** [enthought/mayavi#503]
I've encountered a weird issue where importing mayavi disables printing in the Jupyter notebook. I've currently narrowed down the issue to specifically importing the `import mayavi.tools.helper_functions` module. If I have a notebook with the following cell:

```python
import mayavi.tools.helper_functions
print "hello world"
```

the cell will execute but nothing will print (either from that cell or from any future cell executions). However,  if I do

```python
import mayavi.tools.helper_functions
1 + 1
```

I get `Out[1]: 2`. So it seems to be something specific about the stdout pipe. (I can also reproduce this by writing to `sys.stdout`).

Interestingly this seems to happen only in the notebook and not in terminal IPython.

I am on Ubuntu 16.04 and Python 2.7. Here are versions of relevant libraries;

```bash
[kianilab-cs1 local]{master}$ conda list 'mayavi|jupyter|notebook|ipython'
# packages in environment at /home/mwaskom/anaconda2:
#
ipython                   5.2.2                    py27_0    conda-forge
ipython_genutils          0.1.0                    py27_0  
jupyter                   1.0.0                    py27_3  
jupyter_client            4.3.0                    py27_0  
jupyter_console           4.1.1                    py27_0  
jupyter_core              4.1.0                    py27_0  
mayavi                    4.5.0               np111py27_1    conda-forge
notebook                  5.0.0                    py27_0    conda-forge

```

**Comment by mwaskom** [comment]
A little more information, sometimes when importing mayavi I see the following message through the stderr channel:

```
Traceback (most recent call last):
  File "/home/mwaskom/anaconda2/lib/python2.7/logging/__init__.py", line 882, in emit
    stream.write(fs % msg)
IOError: [Errno 5] Input/output error
Logged from file deprecated.py, line 30
```

**Comment by mwaskom** [comment]
Here's a clue:

```
import sys
sys.stdout
```
```
<ipykernel.iostream.OutStream at 0x7f9fbbafdc50>
```

```
from mayavi import mlab
sys.stdout
```
```
<vtkPythonStdStreamCaptureHelper at 0x7fefb4b2ca38>
```

As a workaround it is possible to capture the stdout stream and reset it after importing mayavi:

```
import sys
stream = sys.stdout
from mayavi import mlab
sys.stdout = stream
```

and then prints work fine.

**Comment by mkoculak** [comment]
Has anyone found a non-workaround solution to this problem?

I have encountered the exact same thing, though my setup is a bit different:
Windows 10, Miniconda 3

```
python                    3.6.3                         0    conda-forge
jupyter_client            5.1.0                    py36_0    conda-forge
jupyter_core              4.4.0                      py_0    conda-forge
notebook                  5.0.0            py36hd9fbf6f_2
mayavi                    4.5.0           np111py36_vc14_2  [vc14]  conda-forge
ipython                   6.2.1                    py36_0    conda-forge
ipython_genutils          0.2.0                    py36_0    conda-forge
```

The workaround helps, but it would be nice to have something more long-lasting.
Or maybe at least have a notice in the mlab + notebook documentation? It took me a lot of time to find this issue, because I thought it is a problem with Jupyter/IPython/IPyKernel.

**Comment by wmvanvliet** [comment]
I narrowed it down further to `tvtk/tools/tvtk_doc.py`, function `get_tvtk_class_names`. This function instantiates all the vtk classes one by one. The `sys.stdout` stream is highjacked by instantiating first the `vtk.vtkPythonInteractiveInterpreter` and then the `vtk.vtkPythonInterpreter` classes. Instantiating one of them, or in reverse order, does not cause any problems, but for some reason, instantiating them both in this order breaks things.

You can reproduce it by starting a fresh kernel and running:

```python
import sys
import vtk
klass = getattr(vtk, 'vtkPythonInteractiveInterpreter')
c = klass()
klass = getattr(vtk, 'vtkPythonInterpreter')
c = klass()
print('Hello??')
print(sys.stdout)
````

**Comment by prabhuramachandran** [comment]
Fixed with #559.
